### PR TITLE
Enable more LRO scenarios

### PR DESCRIPTION
**Pull request by pakrym**
Fixes https://github.com/Azure/autorest.csharp/issues/525 as well.

I've reenabled only the obvious methods to unblock storage.

### CODE REVIEW

File: src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs [hunk 1]
```diff
@@ -304,27 +283,45 @@ private RestClientMethod BuildMethod(Operation operation, string clientName, IRe
...
+                    response.HttpResponse.StatusCodes.Select(ToStatusCode).ToArray(),
+                    responseHeaderModel
+                );
             }
-            else if (response is BinaryResponse)
+            else
             {
-                responseBody = new StreamResponseBody();
+                // Special case for httpInfrastructure testServer swagger, service method always fails
+                clientResponse = new Response(null, Array.Empty<int>(), null);
```
Comment (MiYanni) [inline src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs]: Is this case supposed to pass for that test? I'm confused.

Comment (pakrym) [reply src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs]: This is only to make `httpInfrastructure` weird test swagger work. Nothing else should really be hitting this.

Comment (pakrym) [reply src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs]: I would prefer to have an exception where there is no response but can't.

File: test/TestServerProjects/paging/Generated/Operations/PagingRestClient.cs [hunk 2]
```diff
@@ -1048,11 +1048,7 @@ public async ValueTask<Response<ProductResult>> GetMultiplePagesLROAsync(string
                 switch (message.Response.Status)
                 {
                     case 202:
-                        {
-                            using var document = await JsonDocument.ParseAsync(message.Response.ContentStream, default, cancellationToken).ConfigureAwait(false);
-                            var value = ProductResult.DeserializeProductResult(document.RootElement);
-                            return Response.FromValue(value, message.Response);
-                        }
```
Comment (MiYanni) [inline test/TestServerProjects/paging/Generated/Operations/PagingRestClient.cs]: What made the return type change here? It is still the same status code.

File: src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs (hunk 1 above)
Comment (MiYanni) [reply src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs]: So, the generator fails for `httpInfrastructure` if that isn't there? What changed from before where this wasn't needed?

Comment (pakrym) [reply src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs]: `clientResponse` was nullable and we checked it all the time.

File: test/TestServerProjects/paging/Generated/Operations/PagingRestClient.cs (hunk 2 above)
Comment (pakrym) [reply test/TestServerProjects/paging/Generated/Operations/PagingRestClient.cs]: This is now treated as LRO and not a paging method so we don't care about the initial response type.

**Pull request merged**